libraries/AC_PID/AC_PID.cpp

```diff
@@ -72,109 +86,157 @@ void AC_PID::set_dt(float dt)
     _dt = dt;
 }
 
-// filt_hz - set input filter hz
-void AC_PID::filt_hz(float hz)
+// filt_T_hz - set target filter hz
+void AC_PID::filt_T_hz(float hz)
 {
-    _filt_hz.set(fabsf(hz));
+    _filt_T_hz.set(fabsf(hz));
+}
 
-    // sanity check _filt_hz
-    _filt_hz = MAX(_filt_hz, AC_PID_FILT_HZ_MIN);
+// filt_E_hz - set error filter hz
+void AC_PID::filt_E_hz(float hz)
+{
+    _filt_E_hz.set(fabsf(hz));
+}
+
+// filt_D_hz - set derivative filter hz
+void AC_PID::filt_D_hz(float hz)
+{
+    _filt_D_hz.set(fabsf(hz));
 }
 
-// set_input_filter_all - set input to PID controller
-//  input is filtered before the PID controllers are run
-//  this should be called before any other calls to get_p, get_i or get_d
-void AC_PID::set_input_filter_all(float input)
+//  update_all - set target and measured inputs to PID controller and calculate outputs
+//  target and error are filtered
+//  the derivative is then calculated and filtered
+//  the integral is then updated based on the setting of the limit flag
+float AC_PID::update_all(float target, float measurement, bool limit)
```

Comment:
the name is a bit unintuitive perhaps.  It's an "update" function but it actually returns a value.. but that value isn't really "all" I think because it's missing the FF?